src/format/packed/mod.rs

```diff
@@ -51,18 +42,72 @@ pub enum Variant {
 #[cfg(debug_assertions)]
 impl<T: Element> ::format::Validate for Packed<T> {
     fn validate(&self) {
-        assert_eq!(self.values.len(), storage!(self.size));
+        assert_eq!(self.values.len(), triangular_elems(self.size));
     }
 }
 
 size!(Packed, size, size);
 
+/// Computes the offset into the array for the upper triangular representation
+/// Copied from http://www.netlib.org/lapack/lug/node123.html
+#[inline]
+pub fn upper_triangular_offset(n: usize, i: usize, j: usize) -> usize {
+    debug_assert!(i <= j);
+    debug_assert!(i <= n);
+    debug_assert!(j <= n);
+    // Source material uses 1-based indexing
+    let i = i + 1;
+    let j = j + 1;
+    i + j * (j - 1) / 2 - 1
+}
+
+/// Computes the offset into the array for the lower triangular representation
+/// Copied from http://www.netlib.org/lapack/lug/node123.html
+#[inline]
+pub fn lower_triangular_offset(n: usize, i: usize, j: usize) -> usize {
+    debug_assert!(j <= i);
+    debug_assert!(i <= n);
+    debug_assert!(j <= n);
+    // Source material uses 1-based indexing
```

Comment:
If I also remove this comment, it will be virtually impossible to understand why the formula on the next line works. Are you sure you'd like me to remove the comments that give insight into the math behind why the indexing approach works?